spring-kafka/src/main/java/org/springframework/kafka/support/serializer/DeserializationExceptionHeader.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.support.serializer;
+
+import org.apache.kafka.common.header.internals.RecordHeader;
+
+/**
+ * A package-protected header used to contain serialized
+ * {@link DeserializationException}s. Only headers of this type will be examined for
+ * deserialization.
+ *
+ * @author Gary Russell
+ * @since 2.9.11
```

Comment:
I will back-port here, but I won't back to `2.8.x` until your confirmation.

Reply:
Yeah; let's just leave it at 2.9.x for now; I thought about 2.8.x because it is still in commercial support (2.7.x goes out soon) and it might make it easier if we are asked later to release it. But I don't think it will make much difference doing it now or then.